MHVLib/Clients/MHVRemoteMonitoringClient.m

```diff
@@ -18,61 +18,109 @@
+#import "MHVRestRequest.h"
+#import "MHVServiceResponse.h"
+#import "MHVConnectionFactory.h"
+#import "MHVConnection.h"
+
+@interface MHVRemoteMonitoringClient ()
+
+@property (nonatomic, weak) id<MHVConnectionProtocol>     connection;
+
+@end
 
 @implementation MHVRemoteMonitoringClient
 
-+ (NSURLSessionTask*) requestWithPath:(NSString* _Nonnull)path
-                               method:(NSString* _Nonnull)method
-                           pathParams:(NSDictionary * _Nullable)pathParams
-                          queryParams:(NSDictionary* _Nullable)queryParams
-                           formParams:(NSDictionary * _Nullable)formParams
-                                 body:(id _Nullable)body
+- (instancetype)initWithConnection:(id<MHVConnectionProtocol>)connection
+{
+    MHVASSERT_PARAMETER(connection);
+    
+    self = [super init];
+    if (self)
+    {
+        _connection = connection;
+    }
+    
+    return self;
+}
+
+//PENDING: Swagger generated code needs to update to be injected with RemoteMonitoringClient & remove this method
++ (NSURLSessionTask *)requestWithPath:(NSString *_Nonnull)path
+                               method:(NSString *_Nonnull)method
+                           pathParams:(NSDictionary<NSString *, NSString *> *_Nullable)pathParams
+                          queryParams:(NSDictionary<NSString *, NSString *> *_Nullable)queryParams
+                           formParams:(NSDictionary<NSString *, NSString *> *_Nullable)formParams
+                                 body:(NSData *_Nullable)body
                               toClass:(Class)toClass
-                           completion:(void (^ _Nonnull)(id _Nullable output, NSError * _Nullable error))completion
+                           completion:(void(^_Nonnull)(id _Nullable output, NSError *_Nullable error))completion
 {
+    //    [[MHVRemoteMonitoringClient current] requestWithPath:path
+    //                                              method:method
+    //                                           pathParams:pathParams
+    //                                         queryParams:queryParams
+    //                                          formParams:formParams
+    //                                                body:body
+    //                                             toClass:toClass
+    //                                          completion:completion];
+    
+    return nil;
+}
+
+- (void)requestWithPath:(NSString *_Nonnull)path
+                 method:(NSString *_Nonnull)method
+             pathParams:(NSDictionary<NSString *, NSString *> *_Nullable)pathParams
+            queryParams:(NSDictionary<NSString *, NSString *> *_Nullable)queryParams
+             formParams:(NSDictionary<NSString *, NSString *> *_Nullable)formParams
+                   body:(NSData *_Nullable)body
+                toClass:(Class)toClass
+             completion:(void(^_Nonnull)(id _Nullable output, NSError *_Nullable error))completion
+{
+    if (!completion)
```

Comment:
There may be some cases where consumers of the SDK don't require a completion, we should make the completion optional.